sqlparser/parser.y

```diff
@@ -789,6 +789,12 @@ column_definition_type:
     $1.Autoincrement = BoolVal(true)
     $$ = $1
   }
+| column_definition_type AUTOINCREMENT
+  {
+    $1.Autoincrement = BoolVal(true)
+    $1.ParserMode = ParserModeSQLite3
```

Comment:
`ParserMode` doesn't feel like an attribute of ColumnType. WDYT about adding `AutoIncrement`, using it for `AUTO_INCREMENT`, and using `Autoincrement` for `AUTOINCREMENT` instead?

Reply:
Hmm, but then other implementations would need to deal with both. Let me reconsider it.